compiler/rustc_parse/src/lexer/tokentrees.rs

```diff
@@ -43,10 +43,11 @@ impl<'a> TokenTreesReader<'a> {
                     };
                 }
                 token::Eof => {
-                    if is_delimited {
```

Comment:
Is this always a real EOF, not an end of some token stream in delimiters?
Looks like yes, it this case producing the error seems reasonable.

Reply:
This change is for delay the error emitting to caller, I think mismatch delimiter error comes first seems better, because it's more related to root cause.

And you can see from this commit https://github.com/rust-lang/rust/pull/108297/commits/24117a32c92cd1b5d86b56250d1d72efa62c8789, this change will also remove duplicated diagnostics in test files, but these duplicated ones don't come out in the scenario of command line, I haven't understand the difference between these two.